lib/vsc/utils/script_tools.py

```diff
@@ -58,31 +66,360 @@
+        """
+        self.lockfile = TimestampedPidLockfile(
+            self.options.locking_filename, threshold=self.options.nagios_check_interval_threshold * 2
+        )
+        try:
+            self.lockfile.acquire()
+        except LockFailed as err:
+            raise LockException(f"Failed to acquire lock on {self.options.locking_filename}") from err
+
+    def lock_epilogue(self):
+        """
+        Release the lock on the file
+        """
+        try:
+            self.lockfile.release()
+        except Exception as err:
+            raise LockException("Failed to release lock") from err
+
+
+class TimestampMixin:
+    """
+    A mixin class providing methods for timestamp handling.
+
+    Requires:
+        - The inheriting class must provide `self.options` with attributes:
+            - `start_timestamp`
+            - `TIMESTAMP_FILE_OPTION`
+    """
+    TIMESTAMP_MIXIN_OPTIONS = {
+        "start_timestamp": ("The timestamp form which to start, otherwise use the cached value", None, "store", None),
+        "timestamp_file": (
+            "Location to cache the start timestamp", None, "store",
+            os.path.join(
+                TIMESTAMP_DIR,
+                TIMESTAMP_FILENAME_TEMPLATE % (_script_name(sys.argv[0]),)
+            )
+        ),
+    }
+
+    def timestamp_prologue(self):
+        """
+        Get start time (from commandline or cache), return current time
+        """
+        try:
+            (start_timestamp, current_time) = retrieve_timestamp_with_default(
+                getattr(self.options, TIMESTAMP_FILE_OPTION),
+                start_timestamp=self.options.start_timestamp,
+                default_timestamp=DEFAULT_TIMESTAMP,
+                delta=-MAX_RTT,  # make the default delta explicit, current_time = now - MAX_RTT seconds
+            )
+        except Exception as err:
+            raise TimestampException("Failed to retrieve timestamp") from err
+
+        logging.info("Using start timestamp %s", start_timestamp)
+        logging.info("Using current time %s", current_time)
+        self.start_timestamp = start_timestamp
+        self.current_time = current_time
+
+    def timestamp_epilogue(self):
+        """
+        Write the new timestamp to the file
+        """
+        try:
+            write_timestamp(self.options.timestamp_file, self.current_time)
+        except Exception as err:
+            raise TimestampException("Failed to write timestamp") from err
+
+class LogMixin:
+    """
+    A mixin class providing methods for logging.
+    """
+    LOG_MIXIN_OPTIONS = {
+        'debug': ("Enable debug log mode", None, "store_true", False),
+        'info': ("Enable info log mode", None, "store_true", False),
+        'quiet': ("Enable quiet/warning log mode", None, "store_true", False),
+    }
+
+    def log_prologue(self):
+        """
+        Set the log level
+        """
+        if self.options.quiet:
+            logging.basicConfig(level=logging.WARNING)
```

Comment:
I see a mixture of fancylogger and regular logging, most of the time that does not work (e.g. one or the other get's lost.) has this been tested?